src/OrleansRuntime/GrainTypeManager/GrainTypeData.cs

```diff
@@ -151,5 +154,11 @@ internal static PlacementStrategy GetPlacementStrategy(Type grainClass, Placemen
 
             return predicate;
         }
+
+        private ParameterInfo[] GetConstructorParameters()
+        {
+            var constructor = this.Type.GetConstructors(BindingFlags.Public | BindingFlags.Instance).FirstOrDefault();
+            return constructor?.GetParameters() ?? EmptyParameters;
+        }
```

Comment:
>Binding to constructor parameters so that we can use attributes in them, IMO no, it doesn't need to be core to Orleans.

Supported pattern for this feature does, IMO, need to be core.  As noted in previous comments, there is nothing stopping users from wiring up facets to constructors any way they want, but when in application logic, but we are not responsible for their patterns.  All of the facets we provide and maintain should, however, use the same pattern and that pattern should be the one we support for users.  I agree that the feature logic as a facet should not be tightly bound to how it's wired up, and we can work on that.

>you are coupling this to GrainTypeData, which has nothing to do with how we use DI to resolve stuff.

This data is static data related to the grain type and a grain specific behavior, so I don't see any issue caching it in the grain type data.  We need to have a deterministic means of constructor selection that is independent of DI container in any case, so I don't see how this is limiting or harmful in any way.

>I'm alluding at whether we want to also provide an alternative IStorageFacet implementation that doesn't do the binding to a constructor via the AttributedStorageFacet.

In general, and this needs more work/thought, the features we expose via facets should, IMO, be independent of how they are wired up.  A storage facet, for example, should be independent of the attribute logic and DI factories that wire it up.  The original design had this separation, and I see nothing about what we're doing so far that prevents this.

>I don't think the storage interface should have the Facet suffix

Real facets should not have the suffix, as I stated in other comments.  Facets in this PR are for test and use the name to associate them with the feature.